backend/src/routes/payments.js

```diff
@@ -42,22 +69,43 @@ const paymentReturn = async (req, res) => {
         const ordernumber = req.query.ORDER_NUMBER;
         const payment = await db.payments.getPaymentByOrderNumber(ordernumber);
         if (!payment) {
-            return res
-                .status(422)
-                .json('Payment details not available in the system.');
+            console.log('Payment details not available in the system.');
+            return res.redirect(`/payment-complete?status=1`);
+        }
+
+        if (payment.payment_status) {
+            console.log('This payment was already processed!.');
+            return res.redirect(`/payment-complete?status=2`);
         }
         await db.payments.updatePaymentStatus(ordernumber, req.query.SETTLED);
         const dbUser = await db.users.getUserById(payment.userId);
         const newBalance = payment.amount + dbUser.balance;
-        await db.reservations.updateUserBalance(dbUser.id, newBalance);
+
+        await sequalize.transaction(async (transaction) => {
+            await db.reservations.updateUserBalance(
+                dbUser.id,
+                newBalance,
+                transaction
+            );
+        });
         const balance = await db.reservations.getUserBalance(dbUser.id);
-        res.redirect(`/payment-complete?ordernumber=${ordernumber}`);
+        let paymentsuccess = {};
+        paymentsuccess.ordernumber = ordernumber;
+        paymentsuccess.balance = balance;
+        paymentsuccess.amount = payment.amount;
+        res.redirect(
+            `/payment-complete?ordernumber=${ordernumber}&amount=${
+                payment.amount
+            }&balance=${balance}&status=0`
+        );
     } else {
-        res.status(422).json(`Payment failed. Please try again later`);
+        console.log(`Payment failed. Please try again later`);
```

Comment:
Please include the error code and unique identifier of order to the `console.error`.

Reply:
corrected